desktop/apps/auction2/components/page/Footer.js

```diff
@@ -0,0 +1,80 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import block from 'bem-cn'
+import renderTemplate from 'desktop/components/react/utils/render_template'
+import { first } from 'underscore'
+
+export default function Footer (props) {
+  const { articles, auction, footerItems } = props
+  const footerItem = first(footerItems)
+  const showArticles = Boolean(articles.length)
+  const showFooterItems = footerItem && !auction.isAuctionPromo()
+  const b = block('auction-footer')
+
+  if (!showArticles) {
+    return null
+  }
+
+  return (
+    <footer
+      className={b({without: false})
+        .mix('auction-page-section')
+        .mix(articles.length ? 'has-articles' : 'has-no-articles')}
+    >
+      { showArticles &&
+        <div className={b('auction-articles')}>
+          { articles.models.map((article, key) => {
+            let articleFigureHTML
+
+            // Serverside
+            if (typeof window === 'undefined') {
+              articleFigureHTML = renderTemplate('desktop/components/article_figure/template.jade', { locals: { article, ...props } })
```

Comment:
For right now I think its best to keep this as simple (and divided) as can be so as to inspire us to refactor in React, but that said I think the react lifecycle approach you described above could be leveraged a bit in situations like this. I'll continue thinking about this.